File: python/lv1/p_y_in_string.py
# 문자열 내 p와 y의 개수

'''
Answer 1.

def solution(s):
    pnum = 0
    ynum = 0

    for i in range(0, len(s)):
        if s[i] == "p" or s[i] == "P":
            pnum += 1
            
        elif s[i] == "y" or s[i] == "Y":
            ynum += 1
            
    return pnum == ynum  # True if pnum == ynum else False
'''

'''
Answer 2.

def solution(s):
    return  s.lower().count('p') == s.lower().count('y')

'''

File: cpp/lv1/p_y_in_string.cpp
//문자열 내 p와 y의 개수

#include <string>
#include <iostream>
using namespace std;

/* Answer 1.
bool solution(string s)
{
    int pnum = 0;
    int ynum = 0;
    
    for(int i = 0; i < s.length(); i++)
    {
        if(s[i] == 'p' || s[i] == 'P')
        {
            pnum++;
        }
        else if(s[i] == 'y' || s[i] == 'Y')
        {
            ynum++;
        }
    }
    
    return pnum == ynum;
} */


/* Answer 2.
bool solution(string s)
{
    int pnum = 0;
    int ynum = 0;
    
    for(int i = 0; i < s.length(); i++)
    {
        s[i] = tolower(s[i]);
        
        if(s[i] == 'p')
        {
            pnum++;
        }
        else if(s[i] == 'y')
        {
            ynum++;
        }
    }
    
    return pnum == ynum;
} */
